### PR TITLE
Fix to #23063 - Query/OData: add test support for models requiring non-standard routing (e.g. "Order Details" table in Northwind)

### LINKED ISSUE #23063: Query/OData: add test support for models requiring non-standard routing (e.g. "Order Details" table in Northwind)
For now our tests only support default routing options. In some cases we need customized routing.

### PR DESCRIPTION
**Pull request by maumar**
Adding custom routing for OrderDetailsController

Fixes #23063

### CODE REVIEW

File: test/EFCore.OData.FunctionalTests/Query/NorthwindODataQueryTestFixture.cs [hunk 1]
```diff
@@ -48,4 +61,41 @@ public override async Task DisposeAsync()
...
+            {
+                var entitySetSegment = (EntitySetSegment)odataPath.Segments[0];
+                if (entitySetSegment.EntitySet.Name == "Order Details")
+                {
+                    controller = "OrderDetails";
+                    action = "Get";
+                }
+            }
+
+            if (controller != null && action != null)
```
Comment (smitpatel) [inline test/EFCore.OData.FunctionalTests/Query/NorthwindODataQueryTestFixture.cs:88]: Move this whole block inside the block where controller/action are getting set so we don't additional checks.

**Comment by maumar** [comment]
updated

File: test/EFCore.OData.FunctionalTests/Query/NorthwindODataQueryTestFixture.cs [hunk 2]
```diff
@@ -48,4 +61,33 @@ public override async Task DisposeAsync()
...
+            }
+
+            if (odataPath.PathTemplate == "~/entityset"
+                && routeContext.HttpContext.Request.Method.Equals("get", StringComparison.OrdinalIgnoreCase)
+                && ((EntitySetSegment)odataPath.Segments[0]).EntitySet.Name == "Order Details")
+            {
+                var controller = "OrderDetails";
+                var action = "Get";
+
+                var actionCollectionProvider =
```
Comment (smitpatel) [inline test/EFCore.OData.FunctionalTests/Query/NorthwindODataQueryTestFixture.cs:82]: Can inline all 3 variables. 😄

Comment (maumar) [reply test/EFCore.OData.FunctionalTests/Query/NorthwindODataQueryTestFixture.cs:82]: done :D

**Review by smitpatel — APPROVED** [review]

**Pull request merged**